test/context_test.js

```diff
@@ -111,4 +114,34 @@ suite('API (context)', function() {
       },
     });
   });
+
+  test('Context entry should be known', async () => {
+    //Create test api
+    var api = new subject({
+      title:        'Test Api',
+      description:  'Another test api',
+      context:      ['prop1', 'prop2'],
+      name:         'test',
+    });
+
+    var value = slugid.v4();
+    let validate = await validator({
+      folder:         path.join(__dirname, 'schemas'),
+      baseUrl:        'http://localhost:4321/',
+    });
+    try {
+      api.router({
+        validator: validate,
+        context: {
+          prop3: 'value3',
+        },
+      });
+    } catch (err) {
+      if (/Context has unexpected property: 'prop3'/.test(err)) {
+        return; //expected error
+      }
+      return;
```

Comment:
This makes the test return success even if the expected error didn't occur.  With this return removed(and the one in the previous test, too), the test doesn't pass.  That's a hint that the solution in `src/api.js` isn't correct, as I suggested 7 days ago. I double-checked this by looking for the string "Context has unexpected property".  It only occurs once in the source code (right here).  So nothing could ever throw that error!